### PR TITLE
Enabling caching for get_constructor_type_hints

### PR DESCRIPTION
**Pull request by jcal-15**
Sorry I messed up #6 . 😞 
I am still figuring out how to rebase in git. At my job I just squash when I commit into the main branch.

### CODE REVIEW

**Review by wchresta — CHANGES_REQUESTED** [review]
Awesome :)

**Comment by jcal-15** [comment]
It seems like it would have been better to have first pushed my changes as a commit and then rebase them after you reviewed them. 

If you want, I can undo the rebase so you can see the changes in last commit easier. They were mostly your feedback but I updated some variable names too. Also, I updated ```test_with_nested_global_dataclass``` and ```test_with_nested_module_dataclass``` a little

**Comment by wchresta** [comment]
Either is fine with me; I'll look at it later today but from a first glance it looks great

File: from_dict/_from_dict.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import sys
 import typing
 import functools
+from dataclasses import dataclass
```
Comment (wchresta) [inline from_dict/_from_dict.py:4]: Unused now?

**Review by wchresta — APPROVED** [review]
Nice!

**Pull request merged**